### ISSUE TITLE
no-unnecessary-type-annotation: false positive with typescript@3.0.0

### ISSUE THREAD

**Issue by ajafff** [ajafff/tslint-consistent-codestyle#82]
```ts
const map: Map<string, string> = new Map();

// could be rewritten as
const map = new Map<string, string>();

// but consider a scenario with type aliases
type DisableMap = Map<string, Array<{pos: number, end: number}>>;
const map: DisableMap = new Map();
```
